src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -440,18 +440,10 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
 
         if (blkNode->OperIs(GT_STORE_OBJ))
         {
-            if (!blkNode->AsObj()->GetLayout()->HasGCPtr())
+            if (!blkNode->AsObj()->GetLayout()->HasGCPtr() || (isDstAddrLocal && (size <= copyBlockUnrollLimit)))
```

Comment:
This looks like a bug: before, if `(isDstAddrLocal && (size <= copyBlockUnrollLimit))`, we would set gtBlkOpGcUnsafe=true. Now, we won't unless `!isSrcAddrLocal`.

So, if both isSrcAddrLocal and isDstAddrLocal are true, we won't disable GC. However, the thing we are copying contains GC pointers in this case, and the GC pointers loaded into temporary registers aren't reported, so we have a GC hole.

Reply:
That's true. I missed that case.